enterprise/server/util/redis/redis_test.go

```diff
@@ -0,0 +1,309 @@
+package redis_test
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	redisutil "github.com/buildbuddy-io/buildbuddy/enterprise/server/util/redis"
+	redis "github.com/go-redis/redis/v8"
+)
+
+func genRedisOptions(
+	protocol string,
+	user string,
+	password string,
+	hostname string,
+	port string,
+	database string,
+) *redis.Options {
+	database_num, err := strconv.Atoi(database)
+	if database != "" && err != nil {
+		return &redis.Options{}
+	}
+	if hostname == "" {
+		hostname = "localhost"
+	}
+	if port == "" {
+		port = "6379"
+	}
+	opt := &redis.Options{
+		Network:  "tcp",
+		Username: user,
+		Password: password,
+		Addr:     net.JoinHostPort(hostname, port),
+		DB:       database_num,
+	}
+	if protocol == "rediss" {
+		opt.TLSConfig = &tls.Config{ServerName: hostname}
+	}
+	return opt
+}
+
+func genRedisURI(
+	protocol string,
+	user string,
+	password string,
+	hostname string,
+	port string,
+	database string,
+) string {
```

Comment:
Nit: same as above -- you can omit the "string" on most of these and then probably collapse this into one line